### PR TITLE
Feature/m2m

### PR DESCRIPTION
**Pull request by chrisseto**

### CODE REVIEW

File: include/query.py [hunk 1]
```diff
@@ -129,85 +116,16 @@ def include(self, *related_names, **kwargs):
 
         return clone
 
+    def count(self):
```
Comment (sloria) [inline include/query.py:127]: 👍

File: include/expressions.py [hunk 2]
```diff
@@ -10,3 +18,178 @@ def __init__(self, *args, **kwargs):
 
     def as_sqlite(self, compiler, connection, **extra_context):
         return self.as_sql(compiler, connection, function='JSON_ARRAY', **extra_context)
+
+
+class IncludeExpressionConstructor(object):
```
Comment (sloria) [inline include/expressions.py:23]: I like this structure.

File: include/util.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def parse_datetime(s):
+    if not s:
+        return s
+
+    if ciso8601:
+        return ciso8601.parse_datetime(s)
+
+    # strptime can't parse +00:00
```
Comment (sloria) [inline include/util.py]: Would it make sense to fall back on `django.utils.dateparse.parse_datetime`? It supports timezone offsets.

File: setup.py [hunk 4]
```diff
@@ -45,6 +45,9 @@ def read(fname):
         'psycopg2',
         'ciso8601',
     ],
+    extras_require={
+        'faster': ['ujson', 'ciso8601']
```
Comment (sloria) [inline setup.py]: If you're making these optional deps, then they should be removed from `install_requires`.

File: tests/test_include.py [hunk 5]
```diff
@@ -146,9 +149,52 @@ def test_nested_same(self, django_assert_num_queries):
...
-    pass
+
+    def test_include(self, django_assert_num_queries):
+        cat1 = factories.CatFactory()
+        siblings = factories.CatFactory.create_batch(5)
+        cat1.siblings.add(*siblings)
+
+        with django_assert_num_queries(1):
+            cat = models.Cat.objects.filter(id=cat1.id).include('siblings').first()
+            for sib in cat.siblings.all():
```
Comment (sloria) [inline tests/test_include.py:162]: It might be worth adding a test that does `cat.siblings.filter(.)` as well.

**Review by sloria — COMMENTED** [review]
Looks good. No major blockers.

File: include/util.py (hunk 3 above)
Comment (chrisseto) [reply include/util.py]: Didn't know that was a utility. 👍

File: tests/test_include.py (hunk 5 above)
Comment (chrisseto) [reply tests/test_include.py:162]: 👍

File: setup.py (hunk 4 above)
Comment (chrisseto) [reply setup.py]: Ah, nice catch.

**Pull request merged**